File: clients/android/NewsBlur/src/com/newsblur/activity/RegisterProgress.java
package com.newsblur.activity;

import android.support.v4.app.FragmentActivity;
import android.os.Bundle;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;

import com.newsblur.R;
import com.newsblur.fragment.RegisterProgressFragment;
import com.newsblur.util.PrefsUtils;

/**
 * Show progress screen while registering request is being processed. This
 * Activity doesn't extend NbActivity because it is one of the few
 * Activities that will be shown while the user is still logged out.
 */
public class RegisterProgress extends FragmentActivity {

	private FragmentManager fragmentManager;
	private String currentTag = "fragment";

	@Override
	protected void onCreate(Bundle bundle) {
        PrefsUtils.applyThemePreference(this);

		super.onCreate(bundle);
		setContentView(R.layout.activity_loginprogress);
		
		fragmentManager = getSupportFragmentManager();
		
		if (fragmentManager.findFragmentByTag(currentTag ) == null) {
			final String username = getIntent().getStringExtra("username");
			final String password = getIntent().getStringExtra("password");
			final String email = getIntent().getStringExtra("email");
			FragmentTransaction transaction = fragmentManager.beginTransaction();
			RegisterProgressFragment register = RegisterProgressFragment.getInstance(username, password, email);
			transaction.add(R.id.login_progress_container, register, currentTag);
			transaction.commit();
		}
	}
	
}

File: clients/android/NewsBlur/src/com/newsblur/activity/RegisterProgress.kt
package com.newsblur.activity

import android.content.Intent
import android.os.Bundle
import android.view.animation.AnimationUtils
import android.widget.Toast
import androidx.fragment.app.FragmentActivity
import androidx.lifecycle.lifecycleScope
import com.newsblur.R
import com.newsblur.databinding.ActivityRegisterProgressBinding
import com.newsblur.network.APIManager
import com.newsblur.util.PrefsUtils
import com.newsblur.util.executeAsyncTask
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject

/**
 * Show progress screen while registering request is being processed. This
 * Activity doesn't extend NbActivity because it is one of the few
 * Activities that will be shown while the user is still logged out.
 */
@AndroidEntryPoint
class RegisterProgress : FragmentActivity() {

    @Inject
    lateinit var apiManager: APIManager

    private lateinit var binding: ActivityRegisterProgressBinding

    override fun onCreate(bundle: Bundle?) {
        PrefsUtils.applyThemePreference(this)
        super.onCreate(bundle)
        binding = ActivityRegisterProgressBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val username = intent.getStringExtra("username")
        val password = intent.getStringExtra("password")
        val email = intent.getStringExtra("email")

        binding.progressLogo.startAnimation(AnimationUtils.loadAnimation(this, R.anim.rotate))

        lifecycleScope.executeAsyncTask(
                doInBackground = {
                    apiManager.signup(username, password, email)
                },
                onPostExecute = {
                    if (it.authenticated) {
                        binding.viewSwitcher.showNext()
                    } else {
                        var errorMessage = it.errorMessage
                        if (errorMessage == null) {
                            errorMessage = resources.getString(R.string.register_message_error)
                        }
                        Toast.makeText(this, errorMessage, Toast.LENGTH_LONG).show()
                        startActivity(Intent(this, Login::class.java))
                    }
                }
        )

        binding.buttonNext.setOnClickListener { next() }
    }

    private operator fun next() {
        val i = Intent(this, AddSocial::class.java)
        startActivity(i)
    }
}